### PR TITLE
fix: Fix Sink Config to respect Fallback

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

File: pkg/apis/numaflow/v1alpha1/mono_vertex_types.go [hunk 1]
```diff
@@ -472,10 +472,9 @@ func (mvspec MonoVertexSpec) buildContainers(req getContainerReq) []corev1.Conta
 	if mvspec.Sink.UDSink != nil { // Only support UDSink for now.
 		containers = append(containers, mvspec.Sink.getUDSinkContainer(req))
 	}
-	if mvspec.Sink.Fallback != nil {
+	if mvspec.Sink.Fallback != nil && mvspec.Sink.Fallback.UDSink != nil {
 		containers = append(containers, mvspec.Sink.getFallbackUDSinkContainer(req))
 	}
```
Comment (vigith) [inline pkg/apis/numaflow/v1alpha1/mono_vertex_types.go:477]: don't we support builtins now?

Comment (yhl25) [reply pkg/apis/numaflow/v1alpha1/mono_vertex_types.go:477]: we do support, this check is to avoid creating sidecar container when the fallback sink is not udsink

File: rust/numaflow-core/src/sink/user_defined.rs [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 use tokio::sync::mpsc;
 use tokio_stream::wrappers::ReceiverStream;
 use tonic::transport::Channel;
 use tonic::{Request, Streaming};
 
+use crate::message::{Message, ResponseFromSink};
+use crate::sink::Sink;
+use crate::Error;
+use crate::Result;
+
```
Comment (vigith) [inline rust/numaflow-core/src/sink/user_defined.rs:12]: revert this change?

File: pkg/apis/numaflow/v1alpha1/mono_vertex_types.go (hunk 1 above)
Comment (vigith) [reply pkg/apis/numaflow/v1alpha1/mono_vertex_types.go:477]: gotcha, this was a bug
